### DISCUSSION TITLE
Pass parameters to constructor

### DISCUSSION THREAD

**Discussion by Peterculazh** [inversify/InversifyJS#1517] [Q&A] [answered]
Hello, I would to figure out how I can pass parameters to constructor during resolving dependency?
Here example of my code where I expect to being dependencies injected and passed args to constructor to make my entity ready to go instead of manually passing data in props.
```
@injectable()
export class UserEntity {
    private _id: string | undefined;
    private _language: Language;

    constructor(
        @inject(IUserRepository) private userRepository: IUserRepository,
        { id, language }: { id?: string, language: Language }
    ) {
        // there may go some validation of incoming data
        this._id = id;
        this._language = language;
    }

    save() {
        if (!this._id) {
            return this.userRepository.update(this.id, this);
        }

        return this.userRepository.create(this);
    }

}
```

**Comment by PodaruDragos** [comment, marked as answer]
Hello, this is really not recommended practice, however you can use 2 approaches.

1. create another `injectable` service that you want to use in `save` where you can get an validate data.
2. create a `factory` to get what you want from the `container` [link](https://github.com/inversify/InversifyJS/blob/master/wiki/factory_injection.md)

Hope this helps.

**Reply by capsulated** [reply]
Hi! https://github.com/inversify/InversifyJS/blob/master/wiki/factory_injection.md - 404 Not found 😢

**Reply by notaphplover** [reply]
Hey @capsulated, consider relying on our current docs instead: https://inversify.io/docs/api/binding-syntax/#tofactory

**Reply by capsulated** [reply]
Yep, develop brach, i find it) @notaphplover thx!
